### ISSUE TITLE
Set the Value from an Function Node

### ISSUE THREAD

**Issue by joba64** [lorenwest/node-red-contrib-state#6]
Sorry for the stupid question. I am new to node.js and node-red.

As I understood:
inside a function node you can get the Value of a state with 
  global.get("state").dummy.value

Is it also possible to set this value ????

**Comment by lorenwest** [comment]
Not a stupid question. I had to look it up myself before answering.

The answer is you can't do it this way:
```
global.get('state').dummy.value = 47;
```
Because that won't trigger the state node to persist it or forward that state to other connected nodes.

If you happen to know the ID of the state node (from the ```info``` tab in the Node-RED editor) - it looks something like this: ```2a2117ce.ab3ee8``` - then you can send a message to it like this:
```
let node = RED.nodes.getNode('2a2117ce.ab3ee8');
node.receive({payload:47});
```

Try it out and reply on the thread if this works for you.

**Comment by joba64** [comment]
[flows.txt](https://github.com/lorenwest/node-red-contrib-state/files/5637613/flows.txt)

It doesn't seem to work (see attached flow).

I get this error in the function node : "TypeError: Cannot read property 'getNode' of undefined"
Any ideas ??

**Comment by lorenwest** [comment]
You're right - that doesn't work.  The RED variable in a function node doesn't have all the elements as the one I'm used to when building nodes.

Try setting the output of your function node to the new state, then connect the output of your function node to the input of your state node:
```
msg.payload = 47;
return msg;
```

**Comment by joba64** [comment]
That's what I already have. 
I just tried to avoid to clutter my workflows with all this "the new state is...".
... and I tried to understand how all this node stuff works. 

Thank you for your help !!

**Comment by johnmckerrell** [comment]
I would also like this, I have flows where I want to be able to feed parameters into a flow and use those parameters to decide what state to save. I can get the previous state so it seems like I should be able to save the state too. I can see that you would probably still need to have a setState node somewhere though to set the whole thing up.

As a workaround I'm thinking I could do this by storing an object in the state and just setting `obj["state1"] = 5` and `obj["state2"] = 6` etc. then save the whole object at the end of the flow but that feels a bit of a cheat.

**Comment by chalojak** [comment]
Btw. using `global.get('state').dummy.value = new_value` is not only not working, but also crashing the whole plugin - once any variable is set like this, no other can be read/written.
It is OK, but it maybe worth mentioning to the main readme section - not to use this way of writing a value using function node.

**Comment by lorenwest** [comment]
Node-RED goes to great lengths to make sure the function node doesn't change the state of any other node in the flow. Only way I know to change state of another node is to publish a payload from the function node and connect it to another node that reads that payload.

Unless you want to be tricky and go through TCP/IP or MQTT or something outside the scope of this package.

I'll leave this thread open a while so others can chime in - possibly with some creative mechanism of setting the state in the state node from within a function node.  It's a complex async operation because it converts data types, enforces min/max values, persists the value to disk, and maintains history so a simple `value = new_value` will never work.
